src/seer_server.erl

```diff
@@ -0,0 +1,183 @@
+-module(seer_server).
+
+-include("seer.hrl").
+
+-behaviour(gen_server).
+
+-export(
+    [
+        init/1,
+        handle_cast/2,
+        handle_call/3,
+        handle_info/2,
+        terminate/2,
+        code_change/3
+    ]
+).
+-export([start_link/0]).
+
+-record(
+    state,
+    {
+        mode
+        ::
+        undefined
+        | stdout
+        | carbon
+        | carbon_offline
+        | carbon_reconnected,
```

Comment:
omg fab is that what you meant by weird formatting?

Reply:
Yep. I think I disagree with a lot of Steamroller's choices.